graph/memstore/quadstore_test.go

```diff
@@ -251,3 +256,42 @@ func TestTransaction(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, st, st2, "Appended a new quad in a failed transaction")
 }
+
+// test multi thread insert and query
+func TestMultiThreadQuery(t *testing.T) {
```

Comment:
Can you please add this test to CI with `-race` flag, so it's clear there's none (and none are introduced in the future)?

Also, any chance to not adding more dependencies? Or using something like `x/sync/errgroup`. No offense, it's just more widely used, and there's a chance we already have it as a transitive dependency.